src/with_alloc.rs

```diff
@@ -33,14 +36,42 @@ use core::iter::FromIterator;
 /// buffer.push(1);
 /// assert_eq!(buffer.to_vec(), vec![42, 1]);
 /// ```
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub struct AllocRingBuffer<T> {
-    buf: Vec<T>,
+    buf: Vec<MaybeUninit<T>>,
     capacity: usize,
     readptr: usize,
     writeptr: usize,
 }
 
+impl<T> Drop for AllocRingBuffer<T> {
+    fn drop(&mut self) {
+        for i in self.drain() {
```

Comment:
does `self.drain().foreach(drop)` work? If so that might be a tad nicer

Reply:
fixed in latest commit